File: src/main/java/com/javarush/island/shevchenko/entities/animals/herbivores/Boar.java
package com.javarush.island.shevchenko.entities.animals.herbivores;


import com.javarush.island.shevchenko.entities.organism.Animal;
import com.javarush.island.shevchenko.entities.organism.interfaces.*;
import com.javarush.island.shevchenko.settings.IslandSettings;
import com.javarush.island.shevchenko.util.Randomizer;

/**
 * Класс Boar представляет существо кабана на острове животных.
 * Кабаны являются всеядными и используют поведение из интерфейса Omnivore для питания и охоты.
 */
public class Boar extends Animal implements Omnivore {
    public Boar() {
        this.aClass = Boar.class;
        this.weight = Randomizer.getRandom(IslandSettings.ANIMAL_PARAMETERS.get(Boar.class)[0] / 2, IslandSettings.ANIMAL_PARAMETERS.get(Boar.class)[0]);
    }
}

File: src/main/kotlin/ru/cooper/island/core/model/animal/herbivores/Boar.kt
package ru.cooper.island.core.model.animal.herbivores

import ru.cooper.island.core.model.animal.util.initAnimal
import ru.cooper.island.core.model.Animal
import ru.cooper.island.core.model.traits.Omnivore

/**
 * Кабан (травоядное/всеядное животное)
 * Boar (herbivorous/omnivorous animal)
 */
class Boar : Animal(), Omnivore {
    init { initAnimal<Boar>() }
}
